pkg/controller/installation/products/threescale/reconciler.go

```
@@ -0,0 +1,594 @@
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	logrus.Infof("%s has successfully deployed", packageName)
+
+	phase, err = r.reconcileRHSSOIntegration(serverClient)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	phase, err = r.reconcileUpdatingAdminDetails(serverClient)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	phase, err = r.reconcileServiceDiscovery(serverClient)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	logrus.Infof("%s has reconciled successfully", packageName)
+	return v1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) reconcileNamespace(serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: r.namespace,
+			Labels: map[string]string{
+				"integreatly": "yes",
+			},
+		},
+	}
+	err := serverClient.Get(context.TODO(), pkgclient.ObjectKey{Name: r.namespace}, ns)
+	if err != nil && !k8serr.IsNotFound(err) {
+		return v1alpha1.PhaseFailed, err
+	}
+
+	if err != nil {
+		logrus.Infof("Namespace %s not present", r.namespace)
+		err := serverClient.Create(context.TODO(), ns)
+		if err != nil {
+			return v1alpha1.PhaseFailed, err
+		}
+		logrus.Infof("%s namespace created", r.namespace)
+	}
+
+	if ns.Status.Phase == v1.NamespaceActive {
+		return v1alpha1.PhaseCompleted, nil
+	}
+
+	return v1alpha1.PhaseInProgress, nil
+}
+
+func (r *Reconciler) reconcileOperator(serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	err := r.mpm.CreateSubscription(
+		serverClient,
+		marketplace.GetOperatorSources().Integreatly,
+		r.namespace,
+		packageName,
+		"integreatly",
+		[]string{r.namespace},
+		coreosv1alpha1.ApprovalAutomatic)
+	if err != nil && !k8serr.IsAlreadyExists(err) {
+		return v1alpha1.PhaseFailed, err
+	}
+
+	ip, err := r.mpm.GetSubscriptionInstallPlan(packageName, r.namespace)
```

This error is never checked?